crates/net/network/src/peers/manager.rs

```diff
@@ -216,6 +218,11 @@ impl PeersManager {
         &mut self,
         addr: IpAddr,
     ) -> Result<(), InboundConnectionError> {
+        if let Some(max_total) = self.connection_info.max_total_peers {
+            if self.peers.len() >= max_total {
+                return Err(InboundConnectionError::ExceedsLimit(max_total));
+            }
+        }
```

Comment:
oh nice find, this check was indeed missing!

though we want something like this for inbound:

https://github.com/paradigmxyz/reth/blob/720815a83465a4c79c0066e466ad53d005cc32c1/crates/net/network/src/peers/manager.rs#L816-L819

Reply:
> hough we want something like this for inbound:

thanks, will add it when i fix my rust analyzer because can't cargo check rn